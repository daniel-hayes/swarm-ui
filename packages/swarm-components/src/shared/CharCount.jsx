// @flow
import * as React from 'react';

type CharProps = {
	maxLength: number,
	charLength?: number,
};

export const getRemainingCharacters = (maxLength: number, charLength: number = 0) => maxLength - charLength;

const CharCount = (props: CharProps) => {
	const {
		maxLength,
		charLength = 0,
		...other
	} = props;

	return (
		<p data-swarm-textarea-char-count className="text--tiny" {...other}>
			{getRemainingCharacters(maxLength, charLength)}
		</p>
	);
};

export default CharCount;
