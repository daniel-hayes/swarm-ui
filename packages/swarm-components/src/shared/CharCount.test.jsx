import React from 'react';
import { shallow } from 'enzyme';
import CharCount, {
    getCharacterCount,
} from './CharCount';

describe('CharCount', () => {
	const testCases = [
		['Default', <CharCount key="0" />],
		['with props', <CharCount key="1" className="foo" data-attribute="foo" />],
	];

	test.each(testCases)('Snapshot: %s', (description, element) => {
		expect(shallow(element)).toMatchSnapshot();
    });
});

describe('getCharacterCount', () => {
	it('returns 0 when no argument provided', () => {
		expect(getCharacterCount()).toEqual(0);
	});

	it('returns the length of a string', () => {
		expect(getCharacterCount('')).toEqual(0);
		expect(getCharacterCount('f')).toEqual(1);
		expect(getCharacterCount('f234')).toEqual(4);
		expect(getCharacterCount('ff ff ff')).toEqual(8);
	});
});
