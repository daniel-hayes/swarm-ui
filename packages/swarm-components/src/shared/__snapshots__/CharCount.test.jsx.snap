// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharCount Snapshot: Default 1`] = `
<p
  className="text--tiny"
  data-swarm-textarea-char-count={true}
>
  80
</p>
`;

exports[`CharCount Snapshot: Default with props 1`] = `
<p
  className="foo"
  data-attribute="foo"
  data-swarm-textarea-char-count={true}
>
  80
</p>
`;

exports[`CharCount Snapshot: Value 1`] = `
<p
  className="text--tiny"
  data-swarm-textarea-char-count={true}
  value="some test value"
>
  80
</p>
`;
